test/helpers/fake-github-api.js

```diff
@@ -0,0 +1,26 @@
+/* jshint node: true */
+
+'use strict';
+
+module.exports = FakeGitHubApi;
+
+function FakeGitHubApi(urlOnSuccess, msgOnError, expectedParams) {
+  this.issues = {
+    create: function(params, done) {
+      var actualString = JSON.stringify(params),
+          expectedString = JSON.stringify(expectedParams);
+
+      if (actualString !== expectedString) {
+        done(new Error([
+            'params did not match expectations:',
+            'expected: ' + expectedString,
+            'actual:   ' + actualString
+          ].join('\n  ')));
+      } else if (urlOnSuccess) {
+        done(null, { url: urlOnSuccess });
+      } else {
+        done(new Error(msgOnError));
+      }
+    }
+  };
+}
```

Comment:
Did you look at [Sinon](http://sinonjs.org/docs/#fakeServer) at all? Not sure if it would reduce the need for hooks in your middleware code.